### PR TITLE
Add Django 4.1 support

### PR DESCRIPTION
**Pull request by adamchainz**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -48,7 +48,7 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install tox==3.24.4
+          pip install tox==3.25.1
```
Comment (adamchainz) [inline .github/workflows/main.yml:51]: upgrading tox as well here

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -76,24 +80,32 @@ jobs:
           python: '3.10'
           allow_failure: false
 
-        - name: py39-dj32-postgres-xdist-coverage
+        - name: py39-dj41-mysql_innodb-coverage
```
Comment (adamchainz) [inline .github/workflows/main.yml:83]: did a little rearranging of order so these are always in descending order by python version, then by django version - they were previously inconsistent

**Review by adamchainz — COMMENTED** [review]
Tests show no new warnings 👍

**Review by hramezani — APPROVED** [review]

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -76,24 +80,32 @@ jobs:
...
           allow_failure: false
 
-        - name: py39-djmain-sqlite-coverage
-          python: 3.9
+        - name: py38-dj32-sqlite-xdist-coverage
+          python: 3.8
+          allow_failure: false
+
+        - name: py310-djmain-sqlite-coverage
+          python: '3.10'
```
Comment (samamorgan) [inline .github/workflows/main.yml:108]: Shouldn't this be a float, not a string?

Comment (adamchainz) [reply .github/workflows/main.yml:108]: No because then it will try to install Python 3.1, which is long dead

**Review by bluetech — APPROVED** [review]
Thanks!

**Comment by adamchainz** [comment]
Thank you for merging.

Would you be open to adding me as a maintainer?

**Comment by bluetech** [comment]
Happy to, you should have received an invite.

**Comment by auvipy** [comment]
hope new maintainer can work towards a new release!

**Comment by adamchainz** [comment]
I'm in, thanks! And 👋 @pelme as other maintainer.

**Comment by adamchainz** [comment]
Accepted, thanks!

**Pull request merged**